### PR TITLE
adding `RequesterPays` snippets 

### PR DESCRIPTION
**Pull request by jabubake**
Pending on resolution : 
https://github.com/GoogleCloudPlatform/google-cloud-java/issues/2392
https://github.com/GoogleCloudPlatform/google-cloud-java/issues/2391

and merge:
https://github.com/GoogleCloudPlatform/google-cloud-java/pull/2387

FYI, @frankyn

### CODE REVIEW

**Comment by frankyn** [comment]
Reviewed for sample consistency and clarity. Thanks Jisha!

**Comment by jabubake** [comment]
@shinfan @vam-google  : hope one of you can quickly review these requester pays snippets.

File: google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITStorageSnippets.java [hunk 1]
```diff
@@ -26,19 +26,22 @@
...
 import com.google.cloud.storage.BlobInfo;
 import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.Bucket.BlobTargetOption;
 import com.google.cloud.storage.Storage;
 import com.google.cloud.storage.StorageException;
 import com.google.cloud.storage.testing.RemoteStorageHelper;
 import com.google.common.collect.Iterators;
 import com.google.common.collect.Sets;
 
+import org.apache.http.annotation.ThreadSafe;
```
[the remark this answers is not in the document]
Comment (shinfan) [reply google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITStorageSnippets.java]: Remove unused imports

**Comment by shinfan** [comment]
LGTM

File: google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java [hunk 2]
```diff
@@ -1042,4 +1045,100 @@ public Acl blobToPublicRead(String bucketName, String blobName, long blobGenerat
...
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    BlobSourceOption option = BlobSourceOption.userProject(projectId);
+
+    // read blob in a single pass
+    byte[] readBytes = storage.readAllBytes(bucketName, srcFilename, option);
+
+    // write out to file
+    PrintStream out = new PrintStream(new FileOutputStream(destFilePath.toFile()));
```
Comment (vam-google) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java]: Just a note.
We now have `Blob.downloadTo(Path)` method, which seems applicable here, but unfortunatelly it has to be updated to `Blob.downloadTo(Path, BlobSourceOption...)` to support this use case. It is a rather small change, and probably should be done together with the remaining changes for the RequeseterPays feature. 
After that is done, please consider using it in samples which show how to download a file.

Comment (frankyn) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java]: Thanks for raising this @vam-google. I'll file an issue for this support.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java [hunk 3]
```diff
@@ -1042,4 +1045,100 @@ public Acl blobToPublicRead(String bucketName, String blobName, long blobGenerat
...
+  /**
+   * Example of downloading a file using Requester pay.
+   */
+  public void downloadFileUsingRequesterPays(String projectId, String bucketName,
+      String srcFilename, Path destFilePath) throws IOException {
+    // [START storage_download_file_requester_pays]
+    // Instantiate a Google Cloud Storage client
+
+    // The project ID to bill
+    // String projectId = "my-billable-project-id";
```
Comment (jabubake) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java]: @vam-google : is this the commenting you referred to ? If so, expected : meant to be descriptive to users on what the variables mean.

**Pull request merged**